src/balanced_tree.jl

```diff
@@ -119,21 +119,21 @@ type BalancedTree23{K, D, Ord <: Ordering}
     depth::Int
     freetreeinds::Array{Int,1}
     freedatainds::Array{Int,1}
-    useddatacells::IntSet 
+    useddatacells::IntSet
     # The next two arrays are used as a workspace by the delete!
     # function.
     deletionchild::Array{Int,1}
     deletionleftkey::Array{K,1}
     function BalancedTree23(ord1::Ord)
-        tree1 = Array(TreeNode{K}, 1)
+        tree1 = Array{TreeNode{K}}(1)
```

Comment:
I don't really see why that is important. It is implied by the number of arguments.